File: src/main/java/net/swamphut/swampium/core/swobject/container/SwObject.java
package net.swamphut.swampium.core.swobject.container;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface SwObject {
    Class<?>[] provide() default {};
}

File: src/main/java/net/swamphut/swampium/core/swobject/container/SwObject.kt
package net.swamphut.swampium.core.swobject.container

import kotlin.reflect.KAnnotatedElement

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class SwObject(
        val name: String = ""
) {
    companion object {
        fun fromElement(annotatedElement: KAnnotatedElement) =
                annotatedElement.annotations.mapNotNull { it as?SwObject }.first()
    }
}
